riscv/ns16550.cc

```diff
@@ -341,7 +341,7 @@ std::string ns16550_generate_dts(const sim_t* sim)
   return s.str();
 }
 
-ns16550_t* ns16550_parse_from_fdt(const void* fdt, const sim_t* sim, reg_t* base)
+ns16550_t* ns16550_parse_from_fdt(const void* fdt, const sim_t* sim, reg_t* base, std::vector<std::string> sargs)
```

Comment:
Pass by const ref, instead of passing by value, which will make an unnecessary copy of all the strings.